openwisp_controller/subnet_division/base/models.py

```diff
@@ -118,13 +119,19 @@ def _validate_master_subnet_consistency(self):
                 }
             )
 
-        # Validate master subnet can accommodate required number of generated subnets
-        if self.number_of_subnets > (2 ** (self.size - master_subnet.prefixlen)):
+        # Ensure that master subnet can accommodate
+        # the required number of generated subnets
+        available = 2 ** (self.size - master_subnet.prefixlen)
+        # Account for the reserved subnet
+        available -= 1
```

Comment:
@pandafy the reserved subnet was not taken into account in this calculation.
Therefore, if a master subnet has only 2 available slots but the number of subnets is 2, the first time the rule will run will try to take 3 slots (1 for the reserved subnet plus the 2 requested) and will fail.
Were you aware of this?

Reply:
Great catch on accounting for the reserved subnet! That's a crucial detail I missed.